pkg/apis/logging/v1/elasticsearch_types.go

```diff
@@ -63,7 +63,7 @@ type ElasticsearchSpec struct {
 	RedundancyPolicy RedundancyPolicyType  `json:"redundancyPolicy"`
 	Nodes            []ElasticsearchNode   `json:"nodes"`
 	Spec             ElasticsearchNodeSpec `json:"nodeSpec"`
-	IndexManagement  *IndexManagementSpec  `json:"indexManagement"`
+	IndexManagement  *IndexManagementSpec  `json:"indexManagement,omitempty"`
```

Comment:
drop `omitempty`

Reply:
I'm not certain we want to drop 'omitempty'.  Doesnt this imply that if its null we well get a key with the value of 'null' instead of no key at all?

Reply:
Actually, since the operator-sdk enables openapi validation, we need to drop `omitempty` but add `// +kubebuilder:validation:Optional` such fields that have formerly been optional but we want to have them with an empty value or else we break our codebase

Reply:
@jcantrill no, this means that if the field isn't populated then it won't display it when you do something like `oc get elasticsearch elasticsearch -o yaml`

Reply:
> I'm not certain we want to drop 'omitempty'. Doesnt this imply that if its null we well get a key with the value of 'null' instead of no key at all?

with `omitempty` if the value is empty/nil then we don't see the key either.

Reply:
I've removed the CRD overwrite from this PR for now. Generated CRDS wit OpenAPI validation need a careful check before merging them. I will put them in a follow-up PR.

Reply:
Follow-up PR #285